runtime/hydradx/src/xcm.rs

```diff
@@ -210,7 +216,7 @@ impl pallet_xcm::Config for Runtime {
 	type SendXcmOrigin = EnsureXcmOrigin<RuntimeOrigin, LocalOriginToLocation>;
 	type XcmRouter = XcmRouter;
 	type ExecuteXcmOrigin = EnsureXcmOrigin<RuntimeOrigin, LocalOriginToLocation>;
-	type XcmExecuteFilter = Nothing;
+	type XcmExecuteFilter = AllowTransferAndSwap<MaxXcmDepth, MaxNumberOfInstructions, RuntimeCall>;
```

Comment:
why is this change necessary? i dont think we are planning to execute xcm messages locally and xcm::execute extrinsic is call filtered anyway currently

Reply:
We want it in combination with changing the call filter to allow UIs to interact with the chain through `pallet_xcm::execute`. This is part of the XCM exchange asset stuff.
Concretely you could imagine our UI swapping tokens on Hydra and then sending them to the destination in one extrinsic/with one XCM.